app/controllers/comments_controller.rb

```diff
@@ -0,0 +1,59 @@
+class CommentsController < ApplicationController
+
+  before_action :authenticate_user!, except: [:show, :index]
+
+  # index
+  def index
+    @comments = Comment.all
+  end
+
+  # new / create new comment part 1
+  def new
+    @post = Post.find(params[:post_id])
+    @comment = @post.comments.build
+  end
+
+  # create / create new comment part 2
+  def create
+    @post = Post.find(params[:post_id])
+    @comment = @post.comments.create!(comment_params)
+    redirect_to post_path(@comment.post)
+  end
+
+  #show
+  def show
+    @post = Post.find(params[:post_id])
+    @comments = Comment.find(params[:id])
+  end
+
+  # edit
+  def edit
+    @post = Post.find(params[:post_id])
```

Comment:
Is there a way to check the comment's post's user to allow who can edit/destroy?